_studio/shared/umc/codec/jpeg_dec/src/jpegdec.cpp

```diff
@@ -2069,23 +2065,19 @@ JERRCODE CJPEGDecoder::UpSampling(uint32_t rowMCU, uint32_t colMCU, uint32_t max
             tmpRoiSize.width = (maxMCU - colMCU) * 8 / 2;
             tmpRoiSize.height = m_curr_scan->mcuHeight / 2;
 
-            status = mfxiSampleDownH2V2_JPEG_8u_C1R(pSrc, srcStep, srcRoiSize, pTmp, tmpStep, tmpRoiSize);
+            status = mfxiSampleDownH2V2_JPEG_8u_C1R(pSrc, srcStep, srcRoiSize, pTmp.get(), tmpStep, tmpRoiSize);
             if(ippStsNoErr != status)
             {
-                free(pTmp);
                 LOG0("Error: mfxiSampleDownH2V2_JPEG_8u_C1R() failed!");
                 return JPEG_ERR_INTERNAL;
             }
 
-            status = mfxsCopy_8u(pTmp, pDst, tmpStep * m_curr_scan->mcuHeight / 2);
+            status = mfxsCopy_8u(pTmp.get(), pDst, tmpStep * m_curr_scan->mcuHeight / 2);
```

Comment:
std::copy should be used instead of mfxsCopy_8u.

Reply:
It's from IPP, also know as ippsCopy_8u.  Are we sure that performance is not affected?

Reply:
Let's discuss this separately.